start

```diff
@@ -142,13 +142,76 @@ haskell_register_ghc_nixpkgs(
 EOF
 )
 
+bindist_zlib=$(cat <<EOF
+http_archive(
+    name = "zlib.dev",
+    build_file_content = """
+load("@rules_cc//cc:defs.bzl", "cc_library")
+cc_library(
+    name = "zlib",
+    # Import ':z' as 'srcs' to enforce the library name 'libz.so'. Otherwise,
+    # Bazel would mangle the library name and e.g. Cabal wouldn't recognize it.
+    srcs = [":z"],
+    hdrs = glob(["*.h"]),
+    includes = ["."],
+    visibility = ["//visibility:public"],
+)
+cc_library(name = "z", srcs = glob(["*.c"]), hdrs = glob(["*.h"]))
+""",
```

Comment:
Using `build_file` instead of `build_file_content` would be more readable. There's no syntax highlighting inside a string. You could store the content in `zlib.BUILD.bazel`.

Reply:
I've moved inlined BUILD content in standalone files (alex.BUILD.bazel and zlib.BUILD.BAZEL) 
 in 8f723ad7bd326ddadea066e252385c1dafa527a8. I've removed alex's genrule too.